### PR TITLE
feat: add feature flags to explorer

### PR DESCRIPTION
**Pull request by seve**
#741 

This PR adds the ability to add and utilize feature flags in explorer.

Use `?spatial=true` for testing, you should see a new button on the menu bar with a train icon

### CODE REVIEW

File: client/src/components/geneExpression/menus/addGenes.tsx
Comment (seve) [inline client/src/components/geneExpression/menus/addGenes.tsx:1]: Unused file

File: client/src/components/app.tsx [hunk 1]
```diff
@@ -35,19 +35,11 @@ interface Props {
...
     dispatch(actions.doInitialDataLoad());
     dispatch(actions.checkExplainNewTab());
     this.forceUpdate();
   }
 
-  _onURLChanged(): void {
-    const { dispatch } = this.props;
-    dispatch({ type: "url changed", url: document.location.href });
-  }
-
```
Comment (seve) [inline client/src/components/app.tsx:50]: This action is doesn't exist

File: client/src/components/menubar/index.tsx [hunk 2]
```diff
@@ -329,6 +334,23 @@ class MenuBar extends React.PureComponent<{}, State> {
               />
             </ButtonGroup>
           ) : null}
+          {isSpatial && (
```
Comment (seve) [inline client/src/components/menubar/index.tsx:337]: temp testing button

File: client/src/components/app.tsx (hunk 1 above)
Comment (tihuan) [reply client/src/components/app.tsx:50]: Thanks for deleting this!

File: client/src/components/geneExpression/quickGene.tsx [hunk 3]
```diff
@@ -206,6 +206,9 @@ function QuickGene() {
               itemDisabled={userDefinedGenesLoading ? () => true : () => false}
               noResults={<MenuItem disabled text="No matching genes." />}
               onItemSelect={(g) => {
+                // (seve): I made an attempt at typing this, but it looks like the type of g is not consistent with the type of the argument in onItemSelect
```
Comment (tihuan) [inline client/src/components/geneExpression/quickGene.tsx:209]: thanks for the comment!

**Review by tihuan — APPROVED** [review]
Looks really great! Thanks for adding this along with all the PR comments, Seve 🙏 Super helpful!

**Pull request merged**